packages/webui/src/components/ui/SourceAndSystemFilter.tsx

```diff
@@ -38,129 +38,93 @@ function SourceAndSystemFilter({ className, ...props }: SourceAndSystemFilterPro
-            <X />
-          </div>
-        )}
-      </IconButton>
+      />
       {isOpen && (
-        <div data-test-id="filter-options" className="absolute right-0 mt-2 z-50">
-          <div className="shadow-lg py-1 rounded-md bg-neutral ring-1 ring-primary">
-            <div className={tw('select-none', 'grid grid-rows-1', dropDownOptionsClasses)}>
-              <div>
-                {hasSystems && (
-                  <div data-test-id="source-items-heading" className="font-bold p-2">
-                    Sources
-                  </div>
-                )}
-                {hasSources ? (
-                  <ul data-test-id="source-items" className="flex flex-col gap-1">
-                    {connections.map(([name, isActive]) => {
-                      const isSelected = filters.sources.includes(name);
-                      const statusColor = isActive ? 'bg-green-300' : 'bg-red-300';
-                      return (
-                        <li
-                          key={name}
-                          data-test-id="source-item"
-                          className="cursor-pointer"
-                          onClick={() => handleSourceSelection(name)}
-                        >
-                          <span
-                            className={tw(
-                              'transition-all p-2 flex flex-row items-center rounded bg-white border border-transparent hover:bg-slate-100',
-                              isSelected && 'bg-slate-100 border-neutral hover:bg-slate-50',
-                            )}
-                          >
-                            <span data-test-id="status" className={tw('w-3 h-3 mr-2 rounded-full', statusColor)}></span>
-                            <span className="flex-1">{name}</span>
-                            {isSelected && <Check className="flex-0 w-6 h-6" />}
-                          </span>
-                        </li>
-                      );
-                    })}
-                  </ul>
-                ) : (
-                  <span data-test-id="no-sources" className="opacity-50 p-2 italic">
-                    No sources connected...
-                  </span>
-                )}
+        <div
+          data-test-id="filter-options"
+          className="absolute right-0 mt-2 z-50 bg-secondary border border-solid border-primary rounded-md p-3 w-96"
+        >
+          <div data-test-id="source-items-heading" className="font-bold p-2 uppercase">
+            Sources
+          </div>
+          {hasSources ? (
+            <ul data-test-id="source-items" className="flex flex-col gap-1">
+              {connections.map(([name, isActive]) => {
+                const isSelected = filters.sources.includes(name);
+                const statusColor = isActive ? 'bg-green-400' : 'bg-red-300';
+                return (
+                  <li
+                    key={name}
+                    data-test-id="source-item"
```

Comment:
do we want these to be enumerated per?

Reply:
It's normally enough to have these share the test id, and then we can both count the number of items, and iterate using `.at(index)`.

Reply:
Oh okay, that makes sense too!